### ISSUE TITLE
last fm unknown album scrobbles lost

### ISSUE THREAD

**Issue by michold** [clementine-player/Clementine#5739]
### System information
- Operating System: MAC OS 10.12.4
- Clementine version: 1.3.1

### Expected behaviour / actual behaviour
Expected behaviour: Clementine scrobbles tracks not tagged with an album
actual behaviour: they aren't scrobbled

### Steps to reproduce the problem (only for bugs)
change album tag of a song to an empty string, save new tag and try scrobbling the song.

I am not sure, but I think it happens because clementine's last fm scrobbler adds "b=%5Bunknown%5D&" to POST song query. After changing album name to, say, 'a', query string changes to "b=a" and scrobbling works just fine. I also noticed iTunes handles the scrobbles of the same tracks fine, and it puts "album=&artist=..." [so album name as empty] in its query string. It might be that last.fm rejects scrobbles with album name set to 'unknown'.

**Comment by liftarn** [comment]
Yes, Last.fm blacklists some track/album/artist names like Unknown.
You obviously have found a workaround.

**Comment by michold** [comment]
Well, the problem is, I want them scrobbled 'albumless', but don't really know C++ on a level good enough for a pull request.

**Comment by logological** [comment]
I find Last.fm's refusal to scrobble tracks with empty album fields annoying, but I would find it even more annoying for the scrobbled album field to be populated with some placeholder value.  I don't think Clementine should be messing with the tags simply to work around a misfeature in a third-party service.

**Comment by liftarn** [comment]
If you tag them using [Picard](https://picard.musicbrainz.org/) the album name is set to [non-album tracks] and it works with Last.fm

**Comment by michold** [comment]
it is not a misfeature. Last.fm scrobbles tracks with empty album field just fine, I did it for years with winamp. Clementine is the one guilty of putting 'Unknown' as album name of these tracks when scrobbling.
